### PR TITLE
Add sampler configuration to instrumentation kind

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Related to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration

### CODE REVIEW

**Comment by pavolloffay** [comment]
@anuraaga would you like to review this one?

File: pkg/instrumentation/sdk.go [hunk 1]
```diff
@@ -61,6 +63,24 @@ func injectCommonSDKConfig(otelinst v1alpha1.Instrumentation, pod corev1.Pod) co
 		})
 	}
 
+	idx = getIndexOfEnv(container.Env, envOTELTracesSampler)
+	// configure sampler only if it is configured in the CR
+	if idx == -1 && otelinst.Spec.Sampler.Type != "" {
+		idxSamplerArg := getIndexOfEnv(container.Env, envOTELTracesSamplerArg)
+		if idxSamplerArg == -1 {
```
Comment (anuraaga) [inline pkg/instrumentation/sdk.go:70]: Hmm - given we know that there is no arg for the default sampler, parent_based_always_on, I think we can be a bit more agressive here and ignore the user's SAMPLER_ARG even if it's set. Or either way, it's probably worth to log a message as it seems like a user error

**Review by anuraaga — APPROVED** [review]

**Comment by jpkrohling** [comment]
This needs a rebase.

File: pkg/instrumentation/sdk.go (hunk 1 above)
Comment (pavolloffay) [reply pkg/instrumentation/sdk.go:70]: I don't think it's worth handling this case. If a user accidentally adds an argument for the always_on it's gonna by ignored by the SDK anyways.

**Comment by pavolloffay** [comment]
rebased

**Review by VineethReddy02 — APPROVED** [review]

**Pull request merged**